assets/pages/new-contributions-landing/contributionsLandingActions.js

```diff
@@ -193,10 +179,43 @@ const sendFormSubmitEventForPayPalRecurring = () =>
     onFormSubmit(formSubmitParameters);
   };
 
-function setValueAndTogglePayPal<T>(setStateValue: T => Action, value: T) {
-  return (dispatch: Function): void => {
+
+function enableOrDisableForm(formIsValid: boolean, state: State, dispatch: Function) {
+
+  const shouldEnable = checkoutFormShouldSubmit({
+    formValid: formIsValid,
+    canContributeWithoutSigningIn: canContributeWithoutSigningIn(
+      state.page.form.contributionType,
+      state.page.user.isSignedIn,
+      state.page.form.userTypeFromIdentityResponse,
+    ),
+    isRecurringContributor: state.page.user.isRecurringContributor,
+  });
+
+  if (shouldEnable && window.enablePayPalButton) {
+    dispatch(setFormIsSubmittable(true));
+    window.enablePayPalButton();
+  } else if (window.disablePayPalButton) {
+    dispatch(setFormIsSubmittable(false));
+    window.disablePayPalButton();
+  }
+}
+
+/*
+  Set the value of a form variable, then set the form validity and submitability
+*/
+function setValue<T>(setStateValue: T => Action, value: T) {
```

Comment:
maybe call this `setFormFieldValue`? `setValue` is a bit general

Reply:
I have just realised why this is called `setValue`. As well as being used to set the state value of all the form fields, it is also used to set the state value of `userTypeFromIdentityResponse`, which isn't a form field. T